### PR TITLE
Change CI to github actions

### PR DESCRIPTION
**Pull request by kharann**

### CODE REVIEW

**Comment by kharann** [comment]
Probably just squash it. I might use this later with deployment to azure

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Continuous Integration
+on:
+  push:
+
+jobs:
+  run-linters:
+    name: Run linters
```
Comment (aslakhol) [inline .github/workflows/ci.yml:7]: Everything is in one job called run-linters.

Comment (kharann) [reply .github/workflows/ci.yml:7]: The action runs them in sepperate jobs. So in total 3 different jobs will be instanced during this

**Comment by oleast** [comment]
Are the steps for `compile typescript` and `prettier` missing?

**Comment by kharann** [comment]
> Are the steps for `compile typescript` and `prettier` missing?

I do believe eslint will run prettier because of the prettier plugin in the eslint config (I also tested this and it will throw errors on prettier errors).

Compile typescript just runs eslint I believe

**Comment by oleast** [comment]
> I do believe eslint will run prettier because of the prettier plugin in the eslint config (I also tested this and it will throw errors on prettier errors).

I'm unsure about this, but it may be.

> Compile typescript just runs eslint I believe

In my experience they will pick up different errors.

**Comment by kharann** [comment]
> > I do believe eslint will run prettier because of the prettier plugin in the eslint config (I also tested this and it will throw errors on prettier errors).
> 
> I'm unsure about this, but it may be.
> 
> > Compile typescript just runs eslint I believe
> 
> In my experience they will pick up different errors.

I took a look at drone and saw it only runs npm run compile-tsc which basically just runs eslint. Am I misunderstanding?

I'll add prettier back to the linter tho

**Comment by oleast** [comment]
> I took a look at drone and saw it only runs npm run compile-tsc which basically just runs eslint. Am I misunderstanding?

Not really. `npm run tsc` will run `tsc --noEmit`, which does a typecheck with the typescript compiler. It will error whenever types are wrong, mismatched or missing.

Eslint will check the code style and usage, but not that the actual types are sound.

There is a small piece of overlap, such as both triggering on unused variables, but they each check their own things as well.

**Review by oleast — APPROVED** [review]

**Pull request merged**